subprojects/gst-plugins-bad/sys/d3d11/gstd3d11compositor.cpp

```diff
@@ -801,6 +822,41 @@ gst_d3d11_compositor_pad_set_property (GObject * object, guint prop_id,
           (GstD3D11CompositorSizingPolicy) g_value_get_enum (value);
       pad->position_updated = TRUE;
       break;
+    case PROP_PAD_CROP:
+    {
+        GST_DEBUG_OBJECT(pad, "setting crop parameters");
```

Comment:
please wrap this code to

```
GST_OBJECT_LOCK (pad->convert);
...
GST_OBJECT_UNLOCK (pad->convert);
```
Otherwise there is a race condition: we modify crop_properties in the user's thread, and read them in the streaming thread (in gst_d3d11_compositor_pad_setup_converter)

Reply:
also because of the gst_d3d11_converter_update_src_rect